components/constellation/constellation.rs

```diff
@@ -1713,6 +1718,21 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         let _ = sender.send(length as u32);
     }
 
+    fn handle_history_state_pushed(&mut self, pipeline_id: PipelineId, state_id: HistoryStateId) {
+        if !self.pipeline_is_in_current_frame(pipeline_id) {
```

Comment:
Might be worth doing this  check after looking up the `frame`, to avoid looking up the frame twice.